miri-script/src/commands.rs

```diff
@@ -526,37 +526,56 @@ impl Command {
     }
 
     fn fmt(flags: Vec<OsString>) -> Result<()> {
+        use itertools::Itertools;
+
         let e = MiriEnv::new()?;
-        let toolchain = &e.toolchain;
+        let toolchain = &e.toolchain[..];
         let config_path = path!(e.miri_dir / "rustfmt.toml");
+        let flags = &flags[..];
+
+        // Format individual crates
+        let crates = [
+            e.miri_dir.clone(),
+            path!(e.miri_dir / "cargo-miri"),
+            path!(e.miri_dir / "miri-script"),
+            path!(e.miri_dir / "test_dependencies"),
+        ];
+        format_crates(&e.sh, crates.into_iter(), toolchain, &config_path, flags)
+            .context("`cargo fmt` failed")?;
+
+        // Format benchmarks
+        let bnch_cg_miri_path = path!(e.miri_dir / "bench-cargo-miri");
+        let benches = std::fs::read_dir(bnch_cg_miri_path)?.map(|entry| entry.unwrap().path());
+        format_crates(&e.sh, benches, toolchain, &config_path, flags)
+            .context("`cargo fmt` failed")?;
+
+        // Format test-cargo-miri
+        let tst_cg_miri_path = path!(e.miri_dir / "test-cargo-miri");
+        let tests = std::fs::read_dir(tst_cg_miri_path)?
+            .map(|entry| entry.unwrap().path())
+            .filter(|pth| pth.is_dir());
+        format_crates(&e.sh, tests, toolchain, &config_path, flags)
+            .context("`cargo fmt` failed")?;
+
+        // Collect each test file to run rustfmt on.
+        let files = WalkDir::new(e.miri_dir.join("tests"))
+            .into_iter()
+            .filter_entry(|entry| {
+                let name = entry.file_name().to_string_lossy();
+                let ty = entry.file_type();
+                if ty.is_file() {
+                    name.ends_with(".rs")
+                } else {
+                    // dir or symlink. skip `target` and `.git`.
+                    &name != "target" && &name != ".git"
+                }
+            })
+            .filter_ok(|item| item.file_type().is_file())
+            .map_ok(|item| item.path().to_owned())
+            .map(Result::unwrap);
```

Comment:
I'd much prefer if we could avoid having to list every single crate in the repo. Just formatting all .rs files means we don't have to worry about keeping this list up-to-date. So can you please go back to the old structure here, and just do the batching?

Reply:
Yeah thinking about it further last night having two different ways doesn't make too much sense.